bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/scheduler/EvaluationBackgroundTask.kt

```diff
@@ -0,0 +1,95 @@
+package io.bucketeer.sdk.android.internal.scheduler
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.SystemClock
+import io.bucketeer.sdk.android.BKTClient
+import io.bucketeer.sdk.android.BKTClientImpl
+import io.bucketeer.sdk.android.internal.logd
+import io.bucketeer.sdk.android.internal.loge
+import io.bucketeer.sdk.android.internal.logi
+
+internal class EvaluationBackgroundTask : BroadcastReceiver() {
+  override fun onReceive(context: Context?, intent: Intent?) {
+    // As we can't serialize logger we need to use BKTClient singleton.
+    // This means
+    // - BKTClient#initialize must be called first
+    // - background polling happens as long as BKTClient singleton lives
+    val client = try {
```

Comment:
At first, I thought we need to save config and user to run background task, but it turned out that we can't save logger.

So currently background task implementation relies on BKTClient singleton.

Do you have any idea? I think this implementation covers basic use cases.

Reply:
If a user interacts with `BKTClient#stringVariation()` and others, then BKTClient should be alive.